husamhilal review #1 of Networking Guide

husamhilal opened the pull request:
<!-- Thank you for submitting a Pull Request. Please fill out the template below.-->
# Overview/Summary

Replace this with a brief description of what this Pull Request fixes, changes, etc.

## This PR fixes/adds/changes/removes

1. *Replace me*
2. *Replace me*
3. *Replace me*

### Breaking Changes

1. *Replace me*
2. *Replace me*

## Testing Evidence

- *Replace this with any testing evidence to show that your Pull Request works/fixes as described and planned (include screenshots, if appropriate)*

## As part of this Pull Request I have

- [ ] Checked for duplicate [Pull Requests](https://github.com/Azure/Enterprise-Scale-for-AVS/pulls)
- [ ] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Enterprise-Scale-for-AVS/tree/main)
- [ ] Performed testing and provided evidence.
- [ ] Updated relevant and associated documentation.

fguerri commented on Network Design Guide/avs-networking-basics.md:
```diff
@@ -1,41 +1,42 @@
-Azure VMware Solution runs on bare-metal VMWare ESXi nodes deployed in Azure datacenters and attached to a physical network. Just like Expressroute circuits allow Azure customers to establish layer-3 connectivity between their physical datacenter networks and Azure VNets, a dedicated Expressroute implementation provides layer-3 connectivity between physical ESXi nodes and Azure VNets. 
-When an Azure VMware Solution Private Cloud is provisioned, an associated Expressroute circuit is also instantiated in a Microsoft-managed subscription. The private cloud’s owner can connect the circuit to one or more Expressroute virtual network gateways in  Azure VNets, by redeeming authorization keys for the circuit (the same procedure used to create connections between Expressroute gateways a ...
+## What is the role played by ExpressRoute?
+Azure VMware Solution runs on bare-metal VMWare vSphere ESXi nodes deployed in Azure datacenters and attached to a physical network. Just like how ExpressRoute circuits allow Azure customers to establish layer-3 connectivity between their physical datacenter networks and Azure VNets, a dedicated ExpressRoute implementation provides layer-3 connectivity between physical ESXi nodes and Azure VNets. ...
```
The official name of VMWare's hypervisor seems to be just "VMWare ESXi": https://www.vmware.com/products/esxi-and-esx.html

husamhilal replied:
@fguerri got it. Sounds good :) I'm okay with reverting it back.

fguerri commented on Network Design Guide/avs-networking-basics.md:
```diff
@@ -1,41 +1,42 @@
-When an Azure VMware Solution Private Cloud is provisioned, an associated Expressroute circuit is also instantiated in a Microsoft-managed subscription. The private cloud’s owner can connect the circuit to one or more Expressroute virtual network gateways in  Azure VNets, by redeeming authorization keys for the circuit (the same procedure used to create connections between Expressroute gateways a ...
+## What is the role played by ExpressRoute?
+Azure VMware Solution runs on bare-metal VMWare vSphere ESXi nodes deployed in Azure datacenters and attached to a physical network. Just like how ExpressRoute circuits allow Azure customers to establish layer-3 connectivity between their physical datacenter networks and Azure VNets, a dedicated ExpressRoute implementation provides layer-3 connectivity between physical ESXi nodes and Azure VNets. ...
+When an Azure VMware Solution Private Cloud is provisioned, an associated ExpressRoute circuit is also instantiated in a Microsoft-managed subscription. The private cloud’s owner can then connect the circuit to one or more ExpressRoute Virtual Network Gateways in Azure VNets, by redeeming authorization keys for the circuit (the same procedure used to create connections between ExpressRoute Gatewa ...
```
Agreed. Let's replace all occurrences of "Expressroute" with "ExpressRoute"

husamhilal replied:
Agreed!

fguerri commented on Network Design Guide/avs-networking-basics.md:
```diff
@@ -1,41 +1,42 @@
-## What is the role played by Expressroute Global Reach?
-An Azure Expressroute Gateway cannot be used to route traffic between on-prem locations connected to it over different circuits. This limitation applies to the Azure VMware Solution dedicated Expressroute implementation too, as shown in the figure below.
+## What is the role played by ExpressRoute Global Reach?
+An Azure ExpressRoute Gateway cannot be used to route traffic between on-premises locations connected to it over different circuits. This limitation applies to the Azure VMware Solution dedicated ExpressRoute implementation too, as shown in the figure below.
```
Agreed. Let's replace all occurrences of "on-prem" with "on-premises"

husamhilal replied:
Sounds a great plan!

fguerri commented on Network Design Guide/avs-networking-basics.md:
```diff
@@ -1,41 +1,42 @@
  
 ![figure3](media/figure3.png) 
-Figure 3. ExpressRoute does not support routing traffic between different circuits connected to the same gateway.
+Figure 3. ExpressRoute does not support routing traffic between different circuits connected to the same Gateway.
```
Here "gateway" is used as a generic term, just like "circuit" in the same sentence. I think it can stay lower-case. I would use the capitalized version only if the official resource name "Virtual Network Gateway" is used....

husamhilal replied:
ah, makes sense. Thanks for the clarification!

fguerri commented on Network Design Guide/readme.md:
```diff
@@ -1,14 +1,14 @@
 This guide covers network design for Azure VMWare Solution. It encompasses four design areas, summarized below.
 
-1. **Connectivity with on-premises datacenters**. Connectivity between Azure VMware Solution private clouds and on-prem sites supports a broad set of use cases, such as HCX migrations, hybrid applications, remote vSphere or NSX-T Data Center administration. Azure VMware Solution supports multiple options for hybrid connectivity, including Azure ExpressRoute circuits and internet-based IPSec VPNs. ...
+1. **Connectivity with on-premises datacenters**. Connectivity between Azure VMware Solution private clouds and on-premises sites supports a broad set of use cases, such as HCX migrations, hybrid applications, remote vCenter or NSX-T Manager administration. Azure VMware Solution supports multiple options for hybrid connectivity, including Azure ExpressRoute circuits and internet-based IPSec VPNs. ...
```
NSX-T Data Center is the name of the software product that you can administer using NSX-T Manager. I would use "NSX-T Manager" in a sentence like "connect/log into NSX-T Manager" but not "administer NSX-T Manager" (you do not administer an administration console). Therefore, I think that the original version is OK.

husamhilal replied:
got it. Thanks for the explanation 👍

fguerri commented on Network Design Guide/readme.md:
```diff
@@ -1,14 +1,14 @@
+1. **Connectivity with on-premises datacenters**. Connectivity between Azure VMware Solution private clouds and on-premises sites supports a broad set of use cases, such as HCX migrations, hybrid applications, remote vCenter or NSX-T Manager administration. Azure VMware Solution supports multiple options for hybrid connectivity, including Azure ExpressRoute circuits and internet-based IPSec VPNs. ...
 
-2. **Connectivity with Azure Virtual Networks**. Azure VMware Solution runs on bare-metal VMware VSphere clusters that can be connected to native Azure Virtual Networks (VNets) through Azure ExpressRoute. ExpressRoute connections between Azure VMware Solution private clouds and Azure VNET enable building applications that span the two environments or using “jump-box” virtual machines in Azure to  ...
+2. **Connectivity with Azure Virtual Networks**. Azure VMware Solution runs on bare-metal VMware vSphere clusters that can be connected to native Azure Virtual Networks (VNets) through Azure ExpressRoute. ExpressRoute connection between Azure VMware Solution private cloud and Azure VNet enables building applications that span the two environments or using “jump-box” virtual machines in Azure IaaS ...
```
There is definitely a typo here, but it is the missing "s" after VNet. The sentence should read "ExpressRoute connections between Azure VMware Solution private clouds and Azure VNETs enable building applications that span the two environments or using “jump-box” virtual machines....."

husamhilal replied:
Yes, now it makes more sense! Thanks @fguerri !

fguerri reviewed (changes requested):
Thank you @husamhilal for reviewing. Please see my observations in the comments and let me know what you think.

husamhilal commented:
> Thank you @husamhilal for reviewing. Please see my observations in the comments and let me know what you think.

Thanks a lot @fguerri for the feedback. I addressed the points you mentioned. So, you can merge if you'd like to. I also added comments to my initial commits, not sure if you got those as feedback for further enhancements.

fguerri commented:
I have not gone through the other comments yet. I will do. For now, merging the changes that you suggested in this PR.

husamhilal commented:
> I have not gone through the other comments yet. I will do. For now, merging the changes that you suggested in this PR.

Sounds a great plan. Thanks @fguerri 👍
